src/main/java/harvard/Task.java

```diff
@@ -0,0 +1,111 @@
+package harvard;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+/**
+ * Represents a task.
+ */
+
+public abstract class Task {
+    protected String type;
+
+    /**
+     * The description of the task.
+     */
+    protected String description;
+    /**
+     * The status of the task.
+     */
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object.
+     * @param description The description of the task.
+     */
+
+    public Task(String description, String type) {
+        this.description = description;
+        this.isDone = false;
+        this.type = type;
+    }
+    /**
+     * Returns the status icon of the task.
+     * @return The status icon of the task.
+     */
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+    /**
+     * Returns the description of the task.
+     * @return The description of the task.
+     */
+    public String getDescription() {
+        return description;
+    }
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    /**
+     * Marks the task as undone.
+     */
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+    /**
+     * Returns the string representation of the task to be written to file.
+     * @return The string representation of the task to be written to file.
+     */
+
+    public abstract String toFileString();
+
+    /**
+     * Returns the task from the file string.
+     * @param fileString
+     * @return The task from the file string.
+     * @throws DukeException
+     */
+
+    public static Task fromFileString(String fileString) throws DukeException {
```

Comment:
methods should start with a verb, and I'm not sure if fromFileString is a good method name, though I can understand what it's supposed to do from the name!

Reply:
Agreed, a method name you can consider is `addTaskFromString` but feel free to name it as you wish!